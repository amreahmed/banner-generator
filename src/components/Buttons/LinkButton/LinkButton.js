import styled from "styled-components"

const LinkButton = styled.a`
  padding: 0.1875rem 1rem;
  background-color: #66e2d5;
  font-size: 1rem;
  border: 0.125rem solid black;
  border-radius: 0.1875rem;
  margin: 0.5rem;
  font-weight: bold;
  text-decoration: none;
  color: black;

  &:hover {
    cursor: pointer;
    opacity: 0.8;
  }

  &:focus {
    outline: transparent;
  }
`

export default LinkButton
