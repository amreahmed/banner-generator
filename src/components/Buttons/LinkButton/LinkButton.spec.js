import React from 'react';
import LinkButton from './LinkButton';
import { render, fireEvent, cleanup } from 'react-testing-library';

afterEach(cleanup)

describe('<LinkButton />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<LinkButton>Click me</LinkButton>);
      expect(container).toMatchSnapshot();
    });
  });

  describe('children prop', () => {
    it('should display Click me as LinkButton children', () => {
      const { getByText } = render(<LinkButton>Click me</LinkButton>);
      expect(getByText("Click me").textContent).toBe('Click me');
    });
  });

  describe('onClick prop', () => {
    it('should call onClick function when LinkButton is clicked', () => {
      const onClickMock = jest.fn();
      const { getByText } = render(<LinkButton onClick={onClickMock}>Click me</LinkButton>);
      fireEvent.click(getByText('Click me'));
      expect(onClickMock).toHaveBeenCalled();
    });
  });
});
