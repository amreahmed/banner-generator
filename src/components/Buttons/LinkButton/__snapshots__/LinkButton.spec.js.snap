// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkButton /> Snapshots should match snapshot 1`] = `
<div>
  <a
    class="sc-bdVaJa gHighK"
  >
    Click me
  </a>
</div>
`;
